core/src/main/java/org/mule/runtime/core/internal/exception/OnErrorPropagateHandler.java

```diff
@@ -76,7 +79,14 @@ public TemplateOnErrorHandler duplicateFor(Location buildFor) {
   }
 
   private boolean isRedeliveryExhausted(Exception exception) {
-    return (exception instanceof MessageRedeliveredException);
+    if (exception instanceof MessagingException) {
+      Optional<Error> error = ((MessagingException) exception).getEvent().getError();
+      if (error.isPresent()) {
+        String errorIdentifier = error.get().getErrorType().getIdentifier();
+        return errorIdentifier.equals(REDELIVERY_EXHAUSTED.getName());
```

Comment:
isn't the `equals` for the ErrorType ok for this? why checking the internal state? (the name)